### DISCUSSION TITLE
XYChart how to position tooltip at bottom.

### DISCUSSION THREAD

**Discussion by MarkLyck** [airbnb/visx#1649] [Q&A]
How can I position my tooltip so it shows up in the X axis using `XYChart`?

In the screenshot below, I managed to make it always appear at the top using `top={0}`, but I would like to move the tooltip outlined in green to where the red box is, and follow the cursor still of course.

<img width="1035" alt="Screenshot 2023-02-06 at 08 49 00" src="https://user-images.githubusercontent.com/6841110/216914845-37e55609-fac5-4512-ae8f-1522ddad9e1d.png">

currently this is rendered using this:

```tsx
<Tooltip<DataPoint>
        // @ts-expect-error - ?
        top={0}
        renderTooltip={({ tooltipData }) => {
          if (!tooltipData) return null
          const date = tooltipData.datumByKey[series[0].dataKey].datum[xField]
          return <span>{dayjs(date).format('HH:mm DD/MM/YYYY')}</span>
        }}
      />
```

I've of course tried `bottom={0}` but that does not work.

I've also tried `style={{position: 'absolute', bottom: 0}}` but that also doesn't seem to work, likely because it's rendered in a portal, so it's entirely off screen.

Any ideas how I can move this to overlay over the X axis instead of the top?

**Comment by williaster** [comment]
Hi @MarkLyck, I think the easiest way to do this is to tap into the powerful `DataContext` that is exposed in `xychart`! Using this context, you can access the `height`, `margin`,  `theme`, and even the computed `d3` `x/yScale`s used by the chart.

For your purpose, here is an example where I just wrapped the `Tooltip` in a component which uses context to set `top={height - margin.top}` and it works as I think you want: 

https://codesandbox.io/s/tooltip-bottom-4blhfs
<img width="681" alt="image" src="https://user-images.githubusercontent.com/4496521/217151645-beb811b0-8152-43b4-9475-206c813a0506.png">


In case it's useful to you or others as a completely different approach, there was a non-xychart question about using axes to trigger a tooltip [here](https://github.com/airbnb/visx/issues/944).

**Reply by MarkLyck** [reply]
Hi @williaster Thanks so much for this reply! I tried this solution and it "almost" works, but it seems to have a small visual bug where the tooltip will appear in the middle of the chart first and then "teleport" to the desired location.

See video: (it's very quick only like 1 frame when moving the mouse around) but feels buggy to the user.


https://user-images.githubusercontent.com/6841110/218854914-c1baeb98-7fb1-46ad-b727-100b020c8355.mp4

Any idea how to solve this? It gets much worse if I try to add an animation to it and it constantly tries to animate from the central position in the chart to the bottom, rather than smoothly moving from 1 bottom position to another as I was hoping it would.
